libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplatesParser.cs

```diff
@@ -61,20 +61,39 @@ public static class TemplatesParser
             ImportResolverDelegate importResolver = null,
             ExpressionParser expressionParser = null)
         {
-            importResolver = importResolver ?? DefaultFileResolver;
-            var lg = new Templates(content: content, id: id, importResolver: importResolver, expressionParser: expressionParser);
+            return InnerParseText(content, id, importResolver, expressionParser);
+        }
 
+        /// <summary>
+        /// Parser to turn lg content into a <see cref="Templates"/> based on the original LGFile.
+        /// </summary>
+        /// <param name="content">Text content contains lg templates.</param>
+        /// <param name="lg">original LGFile.</param>
+        /// <returns>new <see cref="Templates"/> entity.</returns>
+        public static Templates ParseTextWithRef(string content, Templates lg)
```

Comment:
I think this ParseTextwithRef probably is just a speical case of "InnerParseText", 

can we just expand the ParseText method to accept a cache also, and probably we can remove this method?